httpx/models.py

```diff
@@ -87,17 +87,14 @@ def __init__(
     ) -> None:
         if isinstance(url, rfc3986.uri.URIReference):
             self.components = url
+        elif isinstance(url, rfc3986.iri.IRIReference):
```

Comment:
I don't think we should be exposing extra types as part of this pull request.

(In fact I don't think we should be exposing `rfc3986.uri.URIReference` as part of our public interface in the first place - it'd be much better if we could keep that as an internal implementation detail, in order to allow us to completely switch implementations if needed.)

Reply:
See #170